src/decision_model.jl

```diff
@@ -348,3 +365,286 @@ function DecisionStrategy(diagram::InfluenceDiagram, z::OrderedDict{Name, Decisi
+    for index in CartesianIndices(μ_bar_statevars)
+        # μ_bar defined as marginal distribution for μ-variables with one dimension marginalized out
+        @constraint(model, μ_bar_statevars[index] .== dropdims(sum(μ_statevars, dims=findfirst(isequal(name), C_rjt[name])), dims=findfirst(isequal(name), C_rjt[name]))[index])
+    end
+    return μ_bar_statevars
+end
+
+struct μVariable
+    node::Name
+    statevars::Array{VariableRef}
+end
+
+
+function local_consistency_constraint(model::Model, arc::Tuple{Name, Name}, C_rjt::Dict{Name, Vector{Name}}, μVars::Dict{Name, μVariable})
+    intersection = C_rjt[arc[1]] ∩ C_rjt[arc[2]]
+    C1_minus_C2 = Tuple(setdiff(collect(1:length(C_rjt[arc[1]])), indexin(intersection, C_rjt[arc[1]])))
+    C2_minus_C1 = Tuple(setdiff(collect(1:length(C_rjt[arc[2]])), indexin(intersection, C_rjt[arc[2]])))
+    @constraint(model, 
+        dropdims(sum(μVars[arc[1]].statevars, dims=C1_minus_C2), dims=C1_minus_C2) .== 
+        dropdims(sum(μVars[arc[2]].statevars, dims=C2_minus_C1), dims=C2_minus_C1))
+end
+
+
+function factorization_constraints(model::Model, diagram::InfluenceDiagram, name::Name, μ_statevars::Array{VariableRef}, μ_bar_statevars::Array{VariableRef}, z::OrderedDict{Name, DecisionVariable})
+    I_j_mapping_in_cluster = [findfirst(isequal(node), diagram.C_rjt[name]) for node in diagram.I_j[name]] # Map the information set to the variables in the cluster
+    for index in CartesianIndices(μ_bar_statevars)
+        for s_j in 1:length(diagram.States[name])
+            if isa(diagram.Nodes[name], ChanceNode)
+                # μ_{C_v} = μ_{\bar{C}_v}*p
+                @constraint(model, μ_statevars[Tuple(index)...,s_j] == diagram.X[name][Tuple(index)[I_j_mapping_in_cluster]...,s_j]*μ_bar_statevars[index])
+            elseif isa(diagram.Nodes[name], DecisionNode)
+                # μ_{C_v} ≤ z
+                @constraint(model, μ_statevars[Tuple(index)...,s_j] <= z[name].z[Tuple(index)[I_j_mapping_in_cluster]...,s_j])
+            end
+        end
+    end
+end
+
+
+struct RJTVariables
+    data::Dict{Name, μVariable}
+end
+
+"""
+    RJTVariables(model, diagram, z)
+
+Using the influence diagram and decision variables z from DecisionProgramming.jl, adds the 
+variables and constraints of the corresponding RJT model.
+
+# Arguments
+- `model::Model`: JuMP model into which variables are added.
+- `diagram::InfluenceDiagram`: Influence diagram structure.
+- `z::OrderedDict{Name, DecisionVariable}`: Decision variables from `DecisionVariables` function.
+
+# Examples
+```julia
+μ_s = RJTVariables(model, diagram, z)
+```
+"""
+function RJTVariables(model::Model, diagram::InfluenceDiagram, z::OrderedDict{Name, DecisionVariable})
+    # Get the RJT structure
+    # SHOULD WE DEFINE C_rjt and A_rjt AS PART OF THE INFLUENCE DIAGRAM STRUCT OR AS THEIR OWN, MAYBE BETTER TO DEFINE AS PART OF INFLUENCE DIAGRAM IF THERE IS NEVER A NEED
```

Comment:
I think it makes sense for them to be independent

Reply:
If we make them independent, they need to be passed as arguments to several functions, where diagram is already passed as an argument. Is it worth it, what do you think?